ansible/ocs_cleanup.yaml

```diff
@@ -0,0 +1,215 @@
+---
+- hosts: localhost
+  gather_facts: false
+  become: true
+  user: root
+  vars_files: vars/default.yaml
+  roles:
+  - oc_local
+  tasks:
+  - name: Remove OCS
+    when: ocs_enabled == true
+    block:
+    - name: Delete OCS storage cluster object
+      shell: oc delete --ignore-not-found -f {{ working_yamls_dir }}/ocs/ocs-storage-cluster.yaml
+      register: remove_ocs_storage_cluster
+      failed_when: remove_ocs_storage_cluster.stderr != "" and ("not found" not in remove_ocs_storage_cluster.stderr and "no matches for kind" not in remove_ocs_storage_cluster.stderr)
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Make sure OCS ceph-cluster deployments are deleted
+      shell: |
+        for i in $(oc get deployment -n openshift-storage -l app=rook-ceph-osd --no-headers -o name); do
+          oc delete $i -n openshift-storage
+        done
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait until rook-ceph-mon, rook-ceph-osd and non-operator-noobaa pods are removed
+      shell: oc get pods -n openshift-storage
+      retries: 100
+      delay: 15
+      register: ocs_pods_removed
+      until: (ocs_pods_removed.stdout | regex_findall('rook-ceph-mon-') | length) == 0 and (ocs_pods_removed.stdout | regex_findall('rook-ceph-osd-') | length) == 0
+             and (ocs_pods_removed.stdout | regex_findall('noobaa-') | length) <= 1
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Make sure cephobjectstoreuser resources will be successfully removed
+      shell: |
+        for i in $(oc get cephobjectstoreusers --no-headers -o name -n openshift-storage); do 
+          oc patch $i --type='json' -p='[{"op": "replace", "path": "/metadata/finalizers", "value":[]}]' -n openshift-storage; 
+        done
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+    
+    - name: Delete OCS subscription
+      shell: oc delete --ignore-not-found -f {{ working_yamls_dir }}/ocs/ocs-sub.yaml
+      register: remove_ocs_sub
+      failed_when: remove_ocs_sub.stderr != "" and ("not found" not in remove_ocs_sub.stderr and "no matches for kind" not in remove_ocs_sub.stderr)
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait until openshift-storage namespace is removed
```

Comment:
For cleanup why not just rely on deleting the project?

Reply:
Unfortunately I've never seen removing the namespace work without manual intervention.  That's actually what I initially tried in my early bash script approach.  One or more resources inevitably got stuck, leaving the namespace in a permanent `Terminating` state and thus paralyzing the playbook.  The individual tasks I've included here (executed in this order) are just what seem to offer the best chance of avoiding the issue.

Reply:
Sounds good. Thanks for the explanation.